backend/server/src/models/matchModel.ts

```diff
@@ -0,0 +1,81 @@
+import mongoose, { Schema, type Types } from "mongoose";
+
+export type PlayerColor = "red" | "white";
+export type PointType = "men" | "kote" | "do" | "tsuki" | "hansoku";
+export type MatchType = "group" | "playoff";
+
+export interface MatchPoint {
+  type: PointType;
+  timestamp: Date;
+}
+
+export interface MatchPlayer {
+  id: Types.ObjectId;
+  points: MatchPoint[];
+  color: PlayerColor;
+}
+
+export interface Match {
+  id: Types.ObjectId;
+  startTimestamp?: Date;
+  timerStartedTimestamp: Date | null;
+  elapsedTime: number;
+  endTimestamp?: Date;
+  type: MatchType;
+  players: MatchPlayer[];
+  winner?: Types.ObjectId;
+  comment?: string;
+}
+
+const pointSchema = new Schema<MatchPoint>(
+  {
+    type: { type: String, required: true },
+    timestamp: { type: Date, required: true }
+  },
+  { _id: false }
+);
+
+const playerSchema = new Schema<MatchPlayer>({
+  points: {
+    type: [pointSchema],
+    required: true,
+    default: []
+  },
+  color: { type: String, required: true }
+});
+
+playerSchema.set("toObject", {
+  virtuals: true,
+  versionKey: false,
+  transform: (doc, ret) => {
+    delete ret._id;
+  }
+});
+
+const matchSchema = new Schema<Match>({
+  startTimestamp: { type: Date, required: false },
+  timerStartedTimestamp: { type: Date, required: false, default: null },
+  elapsedTime: { type: Number, required: true, default: 0 },
+  endTimestamp: { type: Date, required: false },
+  type: { type: String, required: true },
+  winner: { type: Schema.Types.ObjectId, required: false },
+  players: {
+    type: [playerSchema],
+    required: true,
+    maxlength: 2,
+    minlength: 2
+  },
+  comment: { type: String, required: false }
+});
+
+matchSchema.set("toObject", {
+  virtuals: true,
+  versionKey: false,
+  transform: (doc, ret) => {
+    delete ret._id;
+  }
+});
```

Comment:
These and other similar `toObject` methods just overwrite the `_id` field with `id` and omit the version key. These are mainly when sending data to the client.